### PR TITLE
Issue 1,6,7 : Cleaned up code;Extendable cell;Strike off completed…

### PR DESCRIPTION
**Pull request by shrdev**
… task

### CODE REVIEW

**Comment by shrdev** [comment]
@sgoodwin - I just pushed the code. It has all the functionality. We can strike off the completed checklist but if we add a new cell the attributed text in the previous cell is changing.

**Comment by sgoodwin** [comment]
Close! Look in your AppDelegate, many methods that do nothing.

File: Checklist/ChecklistTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+//
+
+import UIKit
+
+class ChecklistTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var textEnteredLabel: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewCell.swift]: This method does nothing.

File: Checklist/ChecklistTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewCell.swift]: This method does nothing.

File: Checklist/ChecklistTableViewController.swift [hunk 3]
```diff
@@ -61,11 +64,28 @@ class ChecklistTableViewController: UITableViewController {
...
             checklists.remove(at: destinationIndexPath.row)
             checklists.insert(itemToMove, at: sourceIndexPath.row)
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let alert = UIAlertController(title: "", message: "Mark as Completed", preferredStyle: .alert)
+        let cancel = UIAlertAction(title: "cancel", style: .cancel, handler: nil)
+        let ok = UIAlertAction(title: "Ok", style: .default) { (action: UIAlertAction) in
+            let cell = tableView.cellForRow(at: indexPath)
+            let label = cell?.viewWithTag(1000) as! UILabel
```
Comment (sgoodwin) [inline Checklist/ChecklistTableViewController.swift]: viewWithTag is not good. Use properties on your custom cell instead.

viewWithTag causes the system to recursively hunt through all of your UIViews to find one with the matching tag. Also later when you forget what tags you used for what, debugging the weird problems that come up will be a pain.

**Comment by sgoodwin** [comment]
For your next changes, I think it's better if you stick to one task per pull request. Makes it easier to talk about each thing.

**Comment by shrdev** [comment]
@sgoodwin all issues have been completed.

**Comment by sgoodwin** [comment]
Hooray!

**Pull request merged**